### PR TITLE
Convert Periodic to use Amazon2023 runners

### PR DESCRIPTION
**Pull request by ZainRizvi**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: .github/workflows/periodic.yml [hunk 1]
```diff
@@ -64,18 +65,19 @@ jobs:
...
-          { config: "default", shard: 1, num_shards: 5, runner: "linux.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 2, num_shards: 5, runner: "linux.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 3, num_shards: 5, runner: "linux.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 4, num_shards: 5, runner: "linux.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 5, num_shards: 5, runner: "linux.4xlarge.nvidia.gpu" },
-          { config: "nogpu_AVX512", shard: 1, num_shards: 1, runner: "linux.2xlarge" },
-          { config: "nogpu_NO_AVX2", shard: 1, num_shards: 1, runner: "linux.2xlarge" },
-          { config: "jit_legacy", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
+          { config: "default", shard: 1, num_shards: 5, runner: "amz2023.linux.4xlarge.nvidia.gpu" },
+          { config: "default", shard: 2, num_shards: 5, runner: "amz2023linux.4xlarge.nvidia.gpu" },
```
Comment (clee2000) [inline .github/workflows/periodic.yml:74]:
```suggestion
          { config: "default", shard: 2, num_shards: 5, runner: "amz2023.linux.4xlarge.nvidia.gpu" },
```

Comment (ZainRizvi) [reply .github/workflows/periodic.yml]: thanks for the catch

**Review by clee2000 — APPROVED** [review]
Why do some jobs get a runner_prefix and others not?

**Comment by ZainRizvi** [comment]
> Why do some jobs get a runner_prefix and others not?

Because `runner_prefix` is a new pattern added in a different PR, and I did it this way to avoid merge conflicts.  Very soon I expect everything will be converted to use `runner_prefix`

**Review by zxiiro — APPROVED** [review]

**Pull request closed without merging**